packages/@react-spectrum/datepicker/src/DatePicker.tsx

```diff
@@ -132,9 +140,20 @@ export function DatePicker(props: SpectrumDatePickerProps) {
             <Content>
               <Calendar
                 autoFocus
-                value={state.value ? state.value.toDate(getLocalTimeZone()) : null}
-                // @ts-ignore
-                onChange={date => selectDate(toCalendarDate(fromDateToLocal(date)))} />
+                value={state.dateValue}
+                onChange={state.setDateValue} />
+              {showTimeField &&
+                <TimeField
```

Comment:
IMO it feels a bit clunky for a keyboard user to interact with the Datepicker dialog that has a TimeField. To set a date and time, the user has to move and confirm to their desired date -> tab into the TimeField -> set each time segment -> shift tab back to the calendar and hit enter on the date. Would be nice if the user could hit enter when inside the TimeField to confirm the selection of both date and time. 

The user can hit escape to close the dialog, but that feels unintuitive (IMO I would expect escape to reset the changes made, kinda like what it does for native input of type=date)

Reply:
Also applies to the DateRangepicker